Topics/Tech_Stacks/ORM.md

```diff
@@ -0,0 +1,133 @@
+# ORM (Object Relation Mapper)
+
+## Table of Content
+### [Introductions](#introductions-1)
+### [What is an ORM](#what-is-an-orm-1)
+### [Example](#example-1)
+### [Should I use one?](#should-i-use-one-1)
+### [Advantages](#advantages-1)
+### [Disadvantages](#disadvantages-1)
+### [Conclusion](#conclusion-1)
+
+## Introductions:
+Most applications that you will build, in fact most applications that exist today, require some sort of data permanence. Meaning that data is stored somewhere, and is not lost when a site/app is reloaded. We employ databases like MYSQL, PostgreSQL, and MongoDB to ensure that our data can we stored and we can safely interact with it. But how do we do that? There are many database drivers that can allow us to write queries as lines of code directly from within our javascript or python (or whatever language you prefer) scripts. On the other hand, you may consider using an ORM to write your database code!
+
+
+## What is an ORM:
+
+An ORM (Object Relation Mapper) is a software tool that allows for the translation of an entity between its representation in a database (A relation) and object-oriented programming languages (An Object).
+
+As such, they allow us to simply treat all data as objects in our code, and leverage the ORM when we want to store this object in our relational database of choice.
+
+## Example:
+Let's take a simple example, where we want to connect to a database, create a user table, and add a new user to our database. How would that look as SQL Queries compared to an ORM? 
+
+```js
+// First the example using an ORM!
+import {Sequelize, DataTypes} from 'sequelize'
+
+const sequelize = new Sequelize(); //Server details go here
+
+const Student = sequelize.define('Student', {
+    id: {
+        type: DataTypes.INTEGER,
+        primaryKey: true,
+        autoIncrement: true
+    },
+    username: {
+        type: DataTypes.STRING,
+        allowNull: false,
+    },
+    student_number: {
+        type: DataTypes.INTEGER,
+        allowNull: false,
+    },
+});
+
+await Student.sync({force:true}); // Ensures table is created in DB
```

Comment:
It might be helpful to mention briefly (with a comment) why 'await' is used since it comes up a few times in the code.

Reply:
Yeah I agree, although this is not the main focus of the document, i still want to readers to be able to effortlessly follow the code snippets. Ill add this!